### PR TITLE
Secret Handshake approach

### PR DESCRIPTION
**Pull request by bobahop**
Approach for Secret Handshake

### CODE REVIEW

File: exercises/practice/secret-handshake/.approaches/slice-indexes/content.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		if (code & (1 << action)) != 0 {
+			output = append(output, signs[action])
+		}
+	}
+	return output
+}
+```
+
+This approach starts by defining a [slice][slice] of the signals in their normal order.
+A [const] is defined to represent the signal for reversing the order of the other signals.
```
Comment (andrerfcsantos) [inline exercises/practice/secret-handshake/.approaches/slice-indexes/content.md:30]:
```suggestion
A [const][const] is defined to represent the signal for reversing the order of the other signals.
```

File: exercises/practice/secret-handshake/.approaches/slice-indexes/content.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+```go
+if (code & (1 << action)) != 0 {
+  output = append(output, signs[action])
+}
+```
+
+For example, if the number being iterated is `0`, then `1` is shifted left `0` times (so not shifted at all), and the number passed in is ANDed with `00001`.
+If the number passed in is `3`, which is `00011` in binary, then it is ANDed with `00001`.
+`00011` ANDed with `00001` is not equal to `0`, so the signal at the index of the slice of signals is added to the output slice.
+The index used is the number being iterated, which is `0`, so the element at index `0` ("wink`) would be added to the output slice
```
Comment (andrerfcsantos) [inline exercises/practice/secret-handshake/.approaches/slice-indexes/content.md:78]:
```suggestion
The index used is the number being iterated, which is `0`, so the element at index `0` (`"wink"`) would be added to the output slice
```

I guess you meant this?

File: exercises/practice/secret-handshake/.approaches/slice-indexes/content.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+using the [append][append] function.
+
+If the number being iterated is `1`, then `1` is shifted left `1` time, and the number passed in is ANDed with `00010`.
+If the number passed in is `3`, which is `00011` in binary, then it is ANDed with `00010`.
+`00011` ANDed with `00010` is not equal to `0`, so the signal at the index of the slice of signals is added to the output slice.
+The index used is the number being iterated, which is `1`, so the element at index `1` ("double blink`) would be added to the output slice.
```
Comment (andrerfcsantos) [inline exercises/practice/secret-handshake/.approaches/slice-indexes/content.md:84]:
```suggestion
The index used is the number being iterated, which is `1`, so the element at index `1` (`"double blink"`) would be added to the output slice.
```

**Review by andrerfcsantos — COMMENTED** [review]
A couple of typos, but looks great otherwise 👍

**Comment by bobahop** [comment]
@andrerfcsantos Thanks for the good catches! I need to give the Quality Control department in my brain a stern talking to. Should be ready to merge now.

**Review by andrerfcsantos — APPROVED** [review]

**Pull request merged**